src/data/general-cs-quiz.ts

```diff
@@ -1565,6 +1565,38 @@ const generalCSQuiz = [
     Explanation:
       "A Linked List contains nodes and each node has data and a reference to next node so that it can have access to the data of its next node and we can get the information of all nodes by iterating the node until it reaches nullptr.",
     Link: "https://www.freecodecamp.org/news/how-linked-lists-work/"
+  },
+  {
+    Question: "What is the Basis for a Red-Black tree?",
+    Answer: "2-3-4 tree",
+    Distractor1: "AVL Tree",
+    Distractor2: "Red-Black Graphs",
+    Distractor3: "Linked List",
+    Explanation:
+      "A Red Black Tree is based upon a 2-3-4 tree where the red nodes collapse into the black node above them to form a 2-3-4 node.",
+    Link: "https://www.cs.purdue.edu/homes/ayg/CS251/slides/chap13b.pdf"
```

Comment:
we only accept freeCodeCamp resources, official documentation or wikipedia for the link field 

```suggestion
    Link: "https://en.wikipedia.org/wiki/Red%E2%80%93black_tree"
```